tb_plugin/torch_tb_profiler/io/file.py

```diff
@@ -527,14 +523,21 @@ def relpath(path, start):
 def join(path, *paths):
     return get_filesystem(path).join(path, *paths)
 
-def download_file(filename):
+def download_file(file_to_download, save_dir):
     """Downloads the file, returning a temporary path to the file after finishing."""
-    return get_filesystem(filename).download_file(filename)
+    local_file = tempfile.NamedTemporaryFile('w+t', suffix='.%s' % basename(file_to_download), dir=save_dir, delete=False)
```

Comment:
the tempfile.NamedTemporaryFile is also business logic belonging to cache.py

Reply:
updated. thanks